src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -276,24 +276,24 @@ private async Task<IReadOnlyList<RestoreSummary>> PerformNuGetV2RestoreAsync(Pac
 
             if (packageRestoreInputs.RestoringWithSolutionFile)
             {
-                Dictionary<string, string> configToProjectPath = GetPackagesConfigToProjectPath(packageRestoreInputs);
+                Dictionary<string, HashSet<string>> configToProjectPath = GetPackagesConfigToProjectsPath(packageRestoreInputs);
```

Comment:
While multiple projects to 1 packages.config is not recommended, it is obviously something that does happen, and I don't want to rock the boat in a patch release, so the fix is to just handle that scenario.

https://github.com/NuGet/Client.Engineering/issues/2851 covers more.